### PR TITLE
Unify buzz methods as MarlinUI::buzz

### PR DESCRIPTION
**Pull request by Ludy87**

### CODE REVIEW

**Comment by Ludy87** [comment]
@thinkyhead  Wait for Feedback

File: Marlin/src/lcd/ultralcd.cpp [hunk 1]
```diff
@@ -595,6 +595,12 @@ void MarlinUI::kill_screen(PGM_P lcd_msg) {
   draw_kill_screen();
 }
 
+#if HAS_BUZZER
+  void MarlinUI::buzz(const long duration, const uint16_t freq) {
+    BUZZ(duration, freq);
+  }
```
Comment (thinkyhead) [inline Marlin/src/lcd/ultralcd.cpp]: This is recursive…
```cpp
#if ENABLED(LCD_USE_I2C_BUZZER)
  #define BUZZ(d,f) ui.buzz(d,f)
  ...
```

File: Marlin/src/lcd/ultralcd.h [hunk 2]
```diff
@@ -259,15 +259,11 @@ class MarlinUI {
...
-    static inline void buzz(const long duration, const uint16_t freq) {
-      #if ENABLED(LCD_USE_I2C_BUZZER)
-        lcd.buzz(duration, freq);
-      #elif PIN_EXISTS(BEEPER)
-        buzzer.tone(duration, freq);
-      #elif ENABLED(PCA9632_BUZZER)
-        pca9632_buzz(duration, freq);
-      #endif
-    }
+    static void buzz(const long duration, const uint16_t freq);
```
Comment (thinkyhead) [inline Marlin/src/lcd/ultralcd.h:262]: `lcd.buzz` and `pca9632_buzz` are lost.

File: Marlin/src/lcd/menu/menu.cpp [hunk 3]
```diff
@@ -441,12 +441,16 @@ void scroll_screen(const uint8_t limit, const bool is_menu) {
...
+      UNUSED(saved);
     #endif
   }
   void lcd_load_settings() {
     const bool loaded = settings.load();
     #if HAS_BUZZER
       ui.completion_feedback(loaded);
+    #else
+      UNUSED(loaded);
     #endif
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu.cpp]: An empty `completion_feedback` method in the case of `!HAS_BUZZER` would clean some of this up.

Comment (Ludy87) [reply Marlin/src/lcd/menu/menu.cpp]: This works only if `buzz` works.
At the moment there are difficulties with LCD, therefore my PR.

File: Marlin/src/lcd/ultralcd.h (hunk 2 above)
Comment (Ludy87) [reply Marlin/src/lcd/ultralcd.h:262]: At the moment it is also very hard to understand code, especially with `ultralcd_ *`

**Comment by Ludy87** [comment]
@thinkyhead I went through some things again.
See: https://github.com/Ludy87/Marlin/commits/bf2_buzzer2

**Comment by thinkyhead** [comment]
This now contains the rebased contents of your `bf2_buzzer2` branch, plus I added a commit taking your changes to their logical extension.

**Comment by timmmoore** [comment]
@thinkyhead, I had to do an edit to this commit to get the pca9632_buzz version to compile. I put the edit in branch https://github.com/timmmoore/Marlin/tree/pca9632_buzz-fixes

**Pull request merged**